### PR TITLE
Amber MMPBSA and MMGBSA for Free energy estimates

### PR DESCRIPTION
**Pull request by chrisbarnettster**
Calculate End-State Free Energies using Ambertools. This uses MMPBSA.py to calculate MMPBSA and MMGBSA

The tests include MMPBSA and decomposition on an Amber tutorial usecase. Entropy and GBSA not included in the tests because these are very slow calculations.

### CODE REVIEW

**Comment by chrisbarnettster** [comment]
@tsenapathi

File: tools/buildtools/ambertools/macros.xml [hunk 1]
```diff
@@ -6,6 +6,9 @@
       <yield/>
     </requirements>
   </xml>
+  <xml name="jinja_requirement">
+     <requirement type="package">jinja2</requirement>
```
Comment (bgruening) [inline tools/buildtools/ambertools/macros.xml]: this needs a version number

File: tools/buildtools/ambertools/mmpbsa_mmgbsa.xml [hunk 2]
```diff
@@ -0,0 +1,238 @@
+<tool id="mmpbsa_mmgbsa" name="mmpbsa mmgbsa" version="@VERSION@">
+  <description>- estimate ligand binding affinities </description>
+  <macros>
+    <import>macros.xml</import>
+  </macros>
+  <expand macro="requirements">
+      <expand macro="jinja_requirement"></expand>
```
Comment (bgruening) [inline tools/buildtools/ambertools/mmpbsa_mmgbsa.xml]: you don't need this ...
you can simply include: `<requirement type="package">jinja2</requirement>`

File: tools/buildtools/ambertools/mmpbsa_mmgbsa.xml [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  </configfiles>
+  <inputs>
+    <section name="input" title="Input" expanded="true">
+      <conditional name="simulation">
+        <param name="simtype" type="select" label="Single or Multiple Trajectories" help="For a single complex in water choose Single. For complex, receptor and ligand trajectories choose multiple">
+          <option selected="True" value="single">Single Trajectory Protocol (STP)</option>
+          <option value="multiple">Multiple Trajectory Protocol (MTP)</option>
+        </param>
+        <when value="single">
+          <param format="txt,prmtop" name="ligand" type="data" label="AMBER prmtop input for Ligand"/>
```
Comment (bgruening) [inline tools/buildtools/ambertools/mmpbsa_mmgbsa.xml]: is prmtop is a valid Galaxy datatype?

File: tools/buildtools/ambertools/mmpbsa_mmgbsa.xml [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        <when value="no">
+        </when>
+      </conditional>
+      <conditional name="decomposition">
+        <param name="decomposition" type="select" label="Decomposition Analysis" help="Choose to carry out decomposition analysis">
+          <option selected="True" value="yes">yes</option>
+          <option value="no">no</option>
+        </param>
+        <when value="yes">
+          <param name="csv_format" type="integer" value="1" label="CSV format" min="0" max="1" help="defaults to yes (1). Choose 0 for unformatted text output"/>
```
Comment (bgruening) [inline tools/buildtools/ambertools/mmpbsa_mmgbsa.xml]: isn't that a boolean?

File: tools/buildtools/ambertools/mmpbsa_mmgbsa.xml [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    <data format="txt" name="resultoutfile" label="${tool.name}: Statistics"/>
+    <data format="txt" name="decompoutfile" label="${tool.name}: Decomposition Statistics"/>
+    <data format="txt" name="parameteroutfile" label="${tool.name}: parameter output"/>
+  </outputs>
+    <tests>
+        <test>
+            <param name="ligand" value="ligand.prmtop" ftype="txt"/>
+            <param name="receptor" value="receptor.prmtop" ftype="txt"/>
+            <param name="complex" value="complex.prmtop" ftype="txt"/>
+	    <param name="trajcomplex" value="1err_desolvated_mini.nc" ftype="netcdf"/>
```
Comment (bgruening) [inline tools/buildtools/ambertools/mmpbsa_mmgbsa.xml]: indentation ...

File: tools/buildtools/ambertools/mmpbsa_mmgbsa.xml [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+            <param name="allparams|entropy" value="0"/>
+            <param name="calcdetails|gbcalc|calctype" value="yes"/>
+            <param name="calcdetails|gbcalc|igb" value="2"/>
```
Comment (bgruening) [inline tools/buildtools/ambertools/mmpbsa_mmgbsa.xml]: there is another syntax, a little bit more verbose but also more readable imho.
```
<conditional name="asd">
   <>
</conditional>
```

File: tools/buildtools/ambertools/mmpbsa_mmgbsa.xml (hunk 3 above)
Comment (chrisbarnettster) [reply tools/buildtools/ambertools/mmpbsa_mmgbsa.xml]: not yet, removing this.

**Comment by chrisbarnettster** [comment]
Thanks @bgruening
Test inputs are from the AMBER tutorial. Made the outputs as small as
possible.
Can try for another usecase?

**Pull request merged**